### DISCUSSION TITLE
How to retain single instance of pages in android.

### DISCUSSION THREAD

**Discussion by DavidM29** [unoplatform/uno#17779] [Q&A] [answered]
Is there a way to make android retain a single instance of pages, like windows does when: `Page.NavigationCacheMode="Enabled"`?

**Comment by FrozDark** [comment]
https://platform.uno/docs/articles/features/native-frame-nav.html#platform-specifics

**Reply by DavidM29** [reply]
I have read this article before and followed its steps, however it is not suitable for my case.

To clarify, I know that pages are preserved on android in the back stack, however navigating forward always creates a new instance of a page, even though I have navigated to that page before. On windows, if the navigation cache is enabled, page instance is created only the first time I navigate to that page, after that it does not matter if I navigate back or forward, this instance is preserved and no new ones will get created.

I would like to have the same approach on android as well.

I am a bit new to uno, so I may be overlooking something or doing something wrong, I am not sure, but this is the conclusion I got from the testing and reading documentations.

**Reply by FrozDark** [reply]
Did you try set `CacheSize` on `Frame` itself?

Did you try to use `FeatureConfiguration.Page.IsPoolingEnabled = true;`?

**Reply by DavidM29** [reply]
Do I need to explicitly set cache size, does the default value of frame cache size differ between platforms? Because, what I am saying works on windows out of the box without specifying the cache size.

I did set `FeatureConfiguration.Page.IsPoolingEnabled = true;`.

**Reply by DavidM29** [reply]
To add some more context.
I have two frames for a bit more complex navigation, the "Shell" frame is inside another frame with separate navigation.

So what the `FeatureConfiguration.Page.IsPoolingEnabled` actually did was reuse a page instance, created in "Frame 1" navigation, in "Frame 2"("Shell Frame") which is not really what I wanted.

Also to clarify, by forward navigation I do not mean `Frame.GoForward()`, I mean navigating to new pages, I probably did not word this well in the initial question.

I achieved what I needed differently, however, it would be nice to know if there is a way to make android navigation work in exactly the same way windows navigation works.

**Reply by jeromelaban** [reply, marked as answer]
Frame is at this time behaving differently from WinUI, wher CacheMode is not supported and pages are always kept in memory, and on android kept in the visual tree so the "back" operation performs better.

We're adjusting this behavior in https://github.com/unoplatform/uno/pull/17615, though it may take time to adjust all platforms.

About the behavior of not creating new pages, this is something that you can manage by "undoing" navigation by clearing the back stack to get to the page that you need. Note that FeatureConfiguration.Page.IsPoolingEnabled is not going to change the current behavior and only is meant to reuse page instances that have been removed from the backstack.
